### PR TITLE
Display address provider via display.json watch

### PR DESCRIPTION
**Pull request by alex-deaconu**
Refactor file watch/retrieve functionality
Update Attribute Data watch to use it

Global eslint config for unit tests

Add pretty task

### CODE REVIEW

**Comment by alex-deaconu** [comment]
@ezequielc please review. @Rise-Vision/content based on the refactoring done, wondering if you could provide review as well. Thanks!

**Comment by stulees** [comment]
@alex-deaconu Agree with @santiagonoguez , you might want to check with Delivery to know if _display.json_ is available to player at the time player injects `getRisePlayerConfiguration()` function into browser window cause the display address could be provided along with the rest of player data.

**Comment by alex-deaconu** [comment]
@santiagonoguez @stulees that was my initial thought as well re: implementation (you can see the first commit).

However, there are a few disadvantages with that approach. First off, it requires player changes. The Electron Player accesses the `display.json` file for Twitter, but the Chrome player does not. And there's no live update for `getRisePlayerConfiguration`.

Based on this, and discussions with @andrecardoso we came up with this solution instead. The main advantage is it works out of the box for each player. Also, it provides live updates for the Display Address when it is updated via `display.json`.

Thanks.

File: src/rise-display-data.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    if ( event.detail.isConnected ) {
+      window.removeEventListener( "rise-local-messaging-connection", connectionHandler );
+
+      _retrieveDisplayData();
+    }
+  }
+
+  const exposedFunctions = {
+    connectionHandler,
+    onDisplayAddress
```
Comment (andrecardoso) [inline src/rise-display-data.js]: As this is a generic display data class, maybe we should also expose the `onDisplayData` method

**Comment by stulees** [comment]
@alex-deaconu makes sense, thanks for clarifying!

File: src/rise-watch.js [hunk 2]
```diff
@@ -1,95 +1,63 @@
...
-  function _handleAttributeDataFileAvailable( fileUrl ) {
+  function _handleFileAvailable( fileUrl, handlerSuccess ) {
 
     return RisePlayerConfiguration.Helpers.getLocalMessagingJsonContent( fileUrl )
       .then( data => {
-        return RisePlayerConfiguration.AttributeData.update( data );
+        return handlerSuccess( data );
       })
       .catch( error => {
         RisePlayerConfiguration.Logger.error(
```
Comment (alex-deaconu) [inline src/rise-watch.js]: @stulees @santiagonoguez one thing that I actually wanted to clarify. As you can see above, for the File Update message errors we call the error handler. However, for File Get errors we just log. I followed the same logic in my refactor, but I'm wondering if this was intentional or I should throw an error here as well. Thanks!

File: src/rise-display-data.js (hunk 1 above)
Comment (alex-deaconu) [reply src/rise-display-data.js]: @andrecardoso actually I intentionally didn't add that getter for the reason that it's not going to be used anywhere at this point (see my comment above https://github.com/Rise-Vision/common-template/pull/58/files#diff-5b579eade4428ffc6924e2e713dd048bR24). 

However, I think you're right, I will update to make it public.

Thanks!

Comment (alex-deaconu) [reply src/rise-display-data.js]: @andrecardoso updated as discussed, see latest commit. Thanks.

**Comment by alex-deaconu** [comment]
To confirm, the functionality for the Display Address was tested on both Chrome OS and Electron (Windows).

Also, I tested the Attribute Data update functionality to ensure the refactoring works as expected.

Merging. Thanks!

**Pull request merged**